feat: add specialised replicated maps

pvlugter opened the pull request:
Refs https://github.com/lightbend/akkaserverless-java-sdk/issues/48. Add specialised replicated maps for counters and registers, similar to Java SDK. Test with the new TCK tests from https://github.com/lightbend/akkaserverless-framework/pull/802 (needs to be published first, draft PR here). And support the new replicated maps from https://github.com/lightbend/akkaserverless-framework/pull/803. Refs https://github.com/lightbend/akkaserverless-java-sdk/issues/49.

octonato commented on sdk/test/replicated-data/register-map-test.js:
```diff
@@ -0,0 +1,176 @@
+      ReplicatedData.createForDelta,
+    );
+    Array.from(registerMap.keys()).should.have.members(['one', 'two', 'three']);
+    registerMap.get('one').should.equal(1);
+    registerMap.get('two').should.equal('foo');
+    registerMap.get('three').field1.should.equal('bar');
```
So, the javascript sdk would allow anything to be added to the register (not only strings) while the java sdk is limited to String. Is that correct? 

Shouldn't we have feature parity?

pvlugter replied:
Both SDKs are the same — keys for maps, and values for registers, can be anything that can be serialized to Any — protobuf messages, json, primitives. Note that the TCK tests expect strings for map keys and register values, but the proxy protocol and SDKs support anything.

pvlugter commented:
Rebased and updated for the TCK in TypeScript now (from #147).

Still needs https://github.com/lightbend/akkaserverless-framework/pull/803 to be released first.

patriknw reviewed (approved):
LGTM

pvlugter commented:
Rebased and updated to framework release 0.7.0-beta.16.
